### ISSUE TITLE
Cannot Find Child In Container

### ISSUE THREAD

**Issue by dschinkel** [airbnb/enzyme#1012]
No matter now many things I try I just cannot find the LoginContainer and I know it's there, I drilled down via WebStorm and see it buried inside HeaderContainer

```
 it.only('only shows logout button when user is authenticated', async () => {
    const store = initStore(),
      headerContainer = mount(<HeaderContainer store={store} />),
      loginContainer = shallow(<LoginContainer store={store} />)

    await mockPost('/users/sessions', { data: [{session: fakeSession, token: fakeToken }]})
    await loginContainer.find('LoginContainer').props().authenticate(fakeEmail, fakePassword)
    let logoutButton = headerContainer.dive().find('LogoutContainer')
    expect(logoutButton).to.exist
  })
```

HeaderContainer
```
import { connect } from 'react-redux'
import React, { Component } from 'react'
import LogoutContainer from '../containers/LogoutContainer'

class HeaderContainer extends Component {
  constructor(props) {
    super(props)

  }

  render(){
    console.log(`this.props.isAuthenticated: ${this.props.isAuthenticated}`)
    return(
      <span>{ this.props.isAuthenticated && <LogoutContainer className='ft-logout-button' /> }</span>
    )
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
})

export default connect(mapStateToProps)(HeaderContainer)
```

of course the expect fails. When I look at this with the WebStorm debugger here's what I'm seeing

<img width="1449" alt="screen shot 2017-06-26 at 7 21 39 pm" src="https://user-images.githubusercontent.com/338798/27565646-17403454-5aa5-11e7-851f-f3083cc79493.png">
<img width="1449" alt="screen shot 2017-06-26 at 7 21 39 pm" src="https://user-images.githubusercontent.com/338798/27565647-1a41cb72-5aa5-11e7-88ad-d3c6e15c3ea7.png">

**Comment by CalMlynarczyk** [comment]
Do a `console.log(headerContainer.debug)` and see what the output is. We recently discovered an issue with ES modules where using default exports was causing Enzyme to not render a component imported as a default for a module with the correct name. I do not know if this issue is caused by Enzyme or something else in the transpilation pipeline.

You might see something like `<Component ... />` in the debug output where you would expect the `<LogoutContainer />` to be. This will cause your `.find('LogoutContainer')` call to not actually find any components matching your selector.

One way to fix this is to declare the `LogoutContainer` component variable first and then export it.

    const LogoutContainer = <div>....</div>;
    export default LogoutContainer;

Enzyme will then render it as a `<LogoutContainer />` component rather than a generic `<Component ... />` component. Again, I did not investigate it further as to why the name of the component gets passed correctly using this method; just a workaround we came across.
